modules/rosa-hcp-aws-privatelink-create-cluster.adoc

```diff
@@ -0,0 +1,93 @@
++
+[IMPORTANT]
+====
+If you use a firewall, you must configure it so that {product-title} can access the sites that it requires to function.
+
+For more information, see the AWS PrivateLink firewall prerequisites section.
+====
+
+. Create the account-wide IAM roles by running the following command:
++
+[source,terminal]
+----
+$ rosa create account-roles --hosted-cp
+----
+
+. Create the OIDC configuration by running the following command:
++
+[source,terminal]
+----
+$ rosa create oidc-config --mode=auto --yes
+----
++
+Save the OIDC configuration ID since you need it to create the operator roles.
++
+.Example output
+[source,terminal]
+----
+I: Setting up managed OIDC configuration
+I: To create Operator Roles for this OIDC Configuration, run the following command and remember to replace <user-defined> with a prefix of your choice:
+	rosa create operator-roles --prefix <user-defined> --oidc-config-id 28s4avcdt2l318r1jbk3ifmimkurk384
+If you are going to create a Hosted Control Plane cluster please include '--hosted-cp'
+I: Creating OIDC provider using 'arn:aws:iam::46545644412:user/user'
+I: Created OIDC provider with ARN 'arn:aws:iam::46545644412:oidc-provider/oidc.op1.openshiftapps.com/28s4avcdt2l318r1jbk3ifmimkurk384'
+----
+
+. Create the Operator roles by running the following command:
++
+[source,terminal]
+----
+$ rosa create operator-roles --hosted-cp --prefix <operator_roles_prefix> --oidc-config-id <oidc_config_id> --installer-role-arn arn:aws:iam::$<account_roles_prefix>:role/$<account_roles_prefix>-HCP-ROSA-Installer-Role
+----
+
+. You can create a private {hcp-title} cluster by running the following command:
++
+[source,terminal]
+----
+$ rosa create cluster --private-link --cluster-name=<cluster-name> --sts --mode=auto --hosted-cp --operator-roles-prefix <operator_role_prefix> --oidc-config-id <oidc_config_id> [--machine-cidr=<VPC CIDR>/16] --subnet-ids=<private-subnet-id1>[,<private-subnet-id2>,<private-subnet-id3>]
+----
+
+. Enter the following command to check the status of your cluster. During cluster creation, the `State` field from the output will transition from `pending` to `installing`, and finally to `ready`.
++
+[source,terminal]
+----
+$ rosa describe cluster --cluster=<cluster_name>
+----
++
+[NOTE]
+====
+If installation fails or the `State` field does not change to `ready` after 10 minutes, check the installation troubleshooting documentation for more details.
```

Comment:
Suggestion to add a link to this doc somewhere in the additional resources, and to reference a specific page title if there is one.